jgrapht-core/src/main/java/org/jgrapht/alg/interval/CircularLinkedList.java

```diff
@@ -0,0 +1,105 @@
+package org.jgrapht.alg.interval;
+
+import java.util.LinkedList;
+import java.util.ListIterator;
+
+/**
+ * The doubly linked circular list
+ *
+ * @param <E> the type of the element stored in the list
+ * @author Jiong Fu
+ */
+public class CircularLinkedList<E> extends LinkedList<E> {
```

Comment:
Unfortunally, this will not work for our approach. This algorithm needs removing elements inside the list. While removing elements from a linked list is generally in O(1), this is not the case for the java LinkedList. Because the java LinkedList does not have any way of interacting with the internal nodes inside the LinkedList, we can not help the linkedList by giving a pointer to the element, which has to be removed. If you want to remove an element from LinkedList, you have to iterate over the whole linkedList to find the element in the first place (even if you use an index!). Thus the complexity for removing in the linkedList is most of the time O(length), which is not duable for our complexity.

Reply:
done by the implementation with CircularListNode.